brownie/project/main.py

```diff
@@ -292,30 +292,116 @@ def _compile_interfaces(self, compiled_hashes: Dict) -> None:
             self._build._add(abi)
 
     def _load_deployments(self) -> None:
-        if CONFIG.network_type != "live":
+        if CONFIG.network_type != "live" and not CONFIG.settings.get("dev_deployment_artifacts"):
             return
-        chainid = CONFIG.active_network["chainid"]
+        chainid = "dev" if CONFIG.network_type != "live" else CONFIG.active_network["chainid"]
         path = self._path.joinpath(f"build/deployments/{chainid}")
         path.mkdir(exist_ok=True)
         deployments = list(path.glob("*.json"))
         deployments.sort(key=lambda k: k.stat().st_mtime)
+        deployment_map: Dict = {}
         for build_json in deployments:
             with build_json.open() as fp:
                 build = json.load(fp)
-            if build["contractName"] not in self._containers:
+
+            contract_name = build["contractName"]
+            if contract_name not in self._containers:
                 build_json.unlink()
                 continue
             if "pcMap" in build:
                 contract = ProjectContract(self, build, build_json.stem)
             else:
                 contract = Contract(  # type: ignore
-                    build["contractName"], build_json.stem, build["abi"]
+                    contract_name, build_json.stem, build["abi"]
                 )
                 contract._project = self
-            container = self._containers[build["contractName"]]
+            container = self._containers[contract_name]
             _add_contract(contract)
             container._contracts.append(contract)
 
+            contract_address = build_json.stem
+            if chainid in deployment_map:
+                if contract_name in deployment_map[chainid] and isinstance(
+                    deployment_map[chainid][contract_name], list
+                ):
+                    deployment_map[chainid][contract_name].insert(0, contract_address)
+                else:
+                    deployment_map[chainid][contract_name] = [contract_address]
```

Comment:
This could all be handled as one line..  the way it is now I find a bit hard to grok.

```python
deployment_map[chainid].setdefault(contract_name, []).insert(0, contract_address)
```

Also, am I correct that inserting at the start means this is sequenced newest to oldest? What is the reasoning there?

Reply:
This is brilliant! I have not heard of setdefault befiore.

It being ordered from newest to oldest is simply for convenience reasons that `contract[0]` is always the newest contract instead of `contract[contract.length - 1]` (yay javascript)

I think the performance impact to shift the list is neglible (and using deque is probably overkill). However, if you think ordering from last to first is more intuitivie I am open to change it

Reply:
no i'm totally fine with it, just didn't see the logic.  I forgot JS doesn't allow negative indexing..  that's a very good reason to prefer to access from the left :)